exporter/azuredataexplorerexporter/metricsdata_to_adx.go

```diff
@@ -0,0 +1,332 @@
+				adxMetrics = append(adxMetrics, &AdxMetric{
+					Timestamp:          dataPoint.Timestamp().AsTime().Format(time.RFC3339),
+					MetricName:         fmt.Sprintf("%s_%s", md.Name(), countsuffix),
+					MetricType:         pmetric.MetricDataTypeSummary.String(), // There is no metric unit for number of / count of
+					MetricDescription:  fmt.Sprintf("%s%s", md.Description(), countdescription),
+					MetricValue:        metricValue,
+					MetricAttributes:   clonedscopeattrs,
+					Host:               host,
+					ResourceAttributes: resourceattrs,
+				})
+			}
+			// now create values for each quantile.
+			for bi := 0; bi < dataPoint.QuantileValues().Len(); bi++ {
+				dp := dataPoint.QuantileValues().At(bi)
+				clonedscopeattrs := cloneMap(scopeattrs)
+				copyAttributes(clonedscopeattrs, dataPoint.Attributes().AsRaw())
+				clonedscopeattrs["qt"] = float64ToDimValue(dp.Quantile())
+				quantilename := fmt.Sprintf("%s_%s", md.Name(), strconv.FormatFloat(dp.Quantile(), 'f', -1, 64))
+				clonedscopeattrs[quantilename] = sanitizeFloat(dp.Value())
+				metricValue := dp.Value()
+				adxMetrics = append(adxMetrics, &AdxMetric{
+					Timestamp:  dataPoint.Timestamp().AsTime().Format(time.RFC3339),
+					MetricName: quantilename,
+					MetricType: pmetric.MetricDataTypeSummary.String(),
+					// There is no unit for a quantile. Will be empty
+					MetricDescription:  fmt.Sprintf("%s%s", md.Description(), countdescription),
+					MetricValue:        metricValue,
+					MetricAttributes:   clonedscopeattrs,
+					Host:               host,
+					ResourceAttributes: resourceattrs,
+				})
+			}
+		}
+		return adxMetrics
+	case pmetric.MetricDataTypeNone:
+		fallthrough
+	default:
+		logger.Warn(
+			"Unsupported metric type : ",
+			zap.Any("metric", md))
+		return nil
+	}
+}
+
+// Given all the metrics , transform that to the representative structure
+func rawMetricsToAdxMetrics(_ context.Context, metrics pmetric.Metrics, logger *zap.Logger) ([]*AdxMetric, error) {
+	var transformedadxmetrics []*AdxMetric
+	resourceMetric := metrics.ResourceMetrics()
+	for i := 0; i < resourceMetric.Len(); i++ {
+		res := resourceMetric.At(i).Resource()
+		scopeMetrics := resourceMetric.At(i).ScopeMetrics()
+		for j := 0; j < scopeMetrics.Len(); j++ {
+			scopemetric := scopeMetrics.At(j)
+			metrics := scopemetric.Metrics()
+			// get details of the scope from the scope metric
+			scopeAttr := getScopeMap(scopemetric.Scope())
+			for k := 0; k < metrics.Len(); k++ {
+				transformedadxmetrics = append(transformedadxmetrics, mapToAdxMetric(res, metrics.At(k), scopeAttr, logger)...)
+			}
+		}
+	}
+	return transformedadxmetrics, nil
+}
+
+func copyAttributes(scopeattrmap map[string]interface{}, metricattributes map[string]interface{}) {
+	for k, v := range metricattributes {
+		scopeattrmap[k] = v
+	}
+}
+
+func cloneMap(fields map[string]interface{}) map[string]interface{} {
+	newFields := make(map[string]interface{}, len(fields))
+	for k, v := range fields {
```

Comment:
you can call copyAttributes here, also probably rename to copyMap to match copyAttributes